src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -170,53 +166,44 @@ public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
   @Override
   public void resourceOffers(SchedulerDriver driver, List<Offer> offers) {
     log.info(String.format("Received %d offers", offers.size()));
-
-    if (liveState.getStagingTasksSize() != 0) {
-      log.info("Declining offers because tasks are currently staging");
-      for (Offer offer : offers) {
+    if (liveState.getCurrentAcquisitionPhase().equals(AcquisitionPhase.RECONCILING_TASKS)
+        && reconciliationComplete()) {
+      correctCurrentPhase();
+    }
+    boolean acceptedOffer = false;
+    for (Offer offer : offers) {
+      if (acceptedOffer) {
         driver.declineOffer(offer.getId());
```

Comment:
This does not fix issue #78 because you still decline offers once accepting a single offer out of the list. Rather, it should be something along the lines of:
```
case JNs:
  if (numRunningJNs + numStagingJNs < numDesiredJNs
      && tryToLaunchJN()) {
      numStagingJNs++;
  } else if (numRunningJNs + numStagingJNs == numDesiredJNs) {
      break; // Launched max numJNs in this offer cycle
  } // else this offer was insufficient, continue to look for better offers
```